glasswall/libraries/rebuild/rebuild.py

```diff
@@ -921,3 +921,81 @@ def GWFileErrorMsg(self):
         error_message = self.library.GWFileErrorMsg()
 
         return error_message
+
+    def _GWFileToFileAnalysisAndProtect(self, input_file: str, file_type: str, output_file: str, output_analysis_report: str):
+        """ This function Manages the specified file and carries out an Analysis Audit, saving the results to the specified file locations.
+
+        Args:
+            input_file (str): The input file path or bytes.
+            output_file (str): The output file path where the protected file will be written.
+            output_analysis_report (str): The output file path where the analysis report will be written.
+
+        Returns:
+            status (int): The result of the Glasswall API call.
+        """
+        # API function declaration
+        self.library.GWFileToFileAnalysisAndProtect.argtypes = [
+            ct.c_wchar_p,  # wchar_t * inputFilePathName
+            ct.c_wchar_p,  # wchar_t* wcType
+            ct.c_wchar_p,  # wchar_t * outputFilePathName
+            ct.c_wchar_p  # wchar_t * analysisFilePathName
+        ]
+
+        # Variable initialisation
+        ct_input_file = ct.c_wchar_p(input_file)
+        ct_file_type = ct.c_wchar_p(file_type)
+        ct_output_file = ct.c_wchar_p(output_file)
+        ct_output_analysis_report = ct.c_wchar_p(output_analysis_report)
+
+        # API call
+        status = self.library.GWFileToFileAnalysisAndProtect(
```

Comment:
Simply use
```
return self.library ...

Reply:
Going to leave this as is to remain consistent with all other API calls, and it makes both the docstring and code more readable to a user.

Reply:
I notice the docstrings are voluminous. Are users aware of their use? Are they displayed in IDEs as pop-ups? The information in them is very useful, but does bloat the code somewhat.

Reply:
> Are they displayed in IDEs as pop-ups?

Yes, as you type in Visual Studio Code for example:

![image](https://github.com/gw-engineering/glasswall-python-wrapper/assets/46464807/8ed2962d-b271-4c86-a942-dad05ea14ba9)